src/organizations/components/OrgProfileTab/index.tsx

```diff
@@ -132,6 +136,44 @@ const OrgProfileTab: FC = () => {
     </FlexBox.Child>
   )
 
+  const LeaveOrgButton = () => {
+    const currentUserId = me.id
+    const {users, handleRemoveUser} = useContext(UsersContext)
+
+    const allowSelfRemoval = users.length > 1
+
+    const handleRemove = () => {
+      handleRemoveUser(currentUserId)
+      window.location.href = CLOUD_URL
+    }
```

Comment:
I know this is copied over from the other file, so don't approach this as a critisism but more as a learning opportunity or chance at gaining perspective. Anyway this method name is backwards and kind of poorly thought out, when you think about it.

Events are handled, whereas actions are performed. The application isn't handling the removal of the user. The application is handling the click event that is generated by the browser when the user clicks the button that says `Leave Organization`.

The action that is performed when then the user clicks `Leave Organization` is to remove the user. The method in the context name is also kind of dumb, because it shows the author's metnal model that has a the tight coupling of the event and the action performed. `removeUser` is a much better name beause it describes what is happening simply - the user is removed. `removeUser` is appropriate to use in response to an event (in this case clicking on the `Leave Organization` button), or if a user needs to be removed by the system as part of some other process that doesn't involve user interaction. Handling the removal of the user sounds like there's an event that is fired by the system called `RemoveUserEvent` and we need to handle it.

The times when it's appropriate to use a method named `handleSomething` are typically when handling an event is a complex process that involves multiple steps or multiple responsibilities. For example, if this component needed to call the method to remove the user and then also call a method to send an email, it might be more appropriate to call this handler `handleClick`, since it's a more generic handler. But in general, I think it's still probably better to encapsulate all the logic to remove a user into a method called `removeUser`, since part of removing the user would also involve sending an email.

It's a small thing from a naming perspective, but a huge thing from an overall system design perspective.

Reply:
> The times when it's appropriate to use a method named `handleSomething` are typically when handling an event is a complex process that involves multiple steps or multiple responsibilities. 

I agree with your assessment, and I appreciate you taking time to explain your points in details. But playing devil's advocate here - I think that's probably why the person who wrote this function named it `handleRemove`. The function is doing two things here. It's invoking the `handleRemoveUser` context by passing it the `currentUserId` and then its redirecting user to `CLOUD_URL`. Since its executing two things here, I'd assume it would make sense to take the `handleSomething` naming convention, no?

Reply:
disregard my comment above. You're saying the function that gets called by the `onConfirm` prop should be named `handleRemoveUser` and inside it, the context should be called `removeUser`. I misunderstood